### PR TITLE
Event-ize preprocess error.

### PR DESCRIPTION
**Pull request by JTCunning**
@mattrobenolt @tkaemming cc @getsentry/platform

---

This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry/4220)

### CODE REVIEW

File: src/sentry/tasks/store.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def preprocess_event(cache_key=None, data=None, start_time=None, **kwargs):
 
     if data is None:
         metrics.incr('events.failed', tags={'reason': 'cache', 'stage': 'pre'})
-        logger.error('Data not available in preprocess_event (cache_key=%s)', cache_key)
+        logger.error('preprocess.failed.empty', extra={'redis_key': cache_key})
```
Comment (mattrobenolt) [inline src/sentry/tasks/store.py]: Why naming this `redis_key`? It's tehcnically not required that this is redis. It's the cache backend, we just happen to use redis ourselves.

Comment (JTCunning) [inline src/sentry/tasks/store.py]: This is the exact reason I asked you what cache this was coming from.

Comment (mattrobenolt) [inline src/sentry/tasks/store.py]: Yeah, our implementation that we run for sentry.io is redis, but someone else could run something else.

**Review by mattrobenolt — APPROVED** [review]
🍕

**Pull request merged**